src/postgres_deparse.c

```diff
@@ -2477,6 +2848,11 @@ static void deparseSelectStmt(DeparseState *state, SelectStmt *stmt)
 	}
 
 	removeTrailingSpace(state);
+
+	if (need_parens || (context != DEPARSE_NODE_CONTEXT_SELECT_SETOP && context != DEPARSE_NODE_CONTEXT_INSERT_SELECT))
+		deparseStateDecreaseNestingLevel(state, parent_level);
```

Comment:
Is there a reason the condition here is more complex than `if (parent_level)`?

Reply:
Yeah, `parent_level` can be NULL - I'll add a comment to clarify.